diff -ru cnijfilter-source-3.80-1/backend/src/cnij_backend_common.c cnijfilter-source/backend/src/cnij_backend_common.c
--- cnijfilter-source-3.80-1/backend/src/cnij_backend_common.c	2012-03-29 06:50:27.000000000 +0200
+++ cnijfilter-source/backend/src/cnij_backend_common.c	2017-10-01 18:15:58.739928638 +0200
@@ -38,6 +38,7 @@
 // CUPS Header
 #include <cups/cups.h>
 #include <cups/ipp.h>
+#include <cups/ppd.h>
 
 // Header file for CANON
 #include "cnij_backend_common.h"
diff -ru cnijfilter-source-3.80-1/backend/src/Makefile.am cnijfilter-source/backend/src/Makefile.am
--- cnijfilter-source-3.80-1/backend/src/Makefile.am	2012-03-29 06:50:27.000000000 +0200
+++ cnijfilter-source/backend/src/Makefile.am	2017-10-01 18:15:58.729928638 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-backend_bindir = /usr/lib/cups/backend
+backend_bindir = /usr/libexec/cups/backend
 backend_bin_PROGRAMS = cnijusb
 transform = 
 
diff -ru cnijfilter-source-3.80-1/backendnet/backend/cnijnetlm.c cnijfilter-source/backendnet/backend/cnijnetlm.c
--- cnijfilter-source-3.80-1/backendnet/backend/cnijnetlm.c	2012-03-29 06:50:28.000000000 +0200
+++ cnijfilter-source/backendnet/backend/cnijnetlm.c	2017-10-01 18:15:58.759928638 +0200
@@ -249,10 +249,9 @@
 	
 	if	( g_pid > 0 ) {				/* if child process has started....	*/
 		int child_status = 0;
-		pid_t child_pid = 0;
 		
 		// waitpid( g_pid, NULL, 0) ;		/* child process wait		*/
-		child_pid = wait(&child_status);
+		wait(&child_status);
 		
 		if (!WIFEXITED(child_status)){
 			return( CANON_STS_NG ) ;
@@ -356,12 +355,11 @@
 			}
 			
 			struct pollfd fds;
-			int pollst;
 			
 			fds.fd = pipe_fds;
 			fds.events = POLLOUT;
 			
-			pollst = poll(&fds, 1, polltime);
+			poll(&fds, 1, polltime);
 			
 			if (fds.revents & POLLOUT){
 				fseek(tempfp, read_position, SEEK_SET);
diff -ru cnijfilter-source-3.80-1/backendnet/backend/Makefile.am cnijfilter-source/backendnet/backend/Makefile.am
--- cnijfilter-source-3.80-1/backendnet/backend/Makefile.am	2012-03-29 06:50:28.000000000 +0200
+++ cnijfilter-source/backendnet/backend/Makefile.am	2017-10-01 18:15:58.749928638 +0200
@@ -1,6 +1,6 @@
 ## Process this file with automake to produce Makefile.in
 
-backendnet_bindir = /usr/lib/cups/backend
+backendnet_bindir = /usr/libexec/cups/backend
 backendnet_bin_PROGRAMS = cnijnet
 transform = 
 
diff -ru cnijfilter-source-3.80-1/backendnet/lmonitor/cnijnetchk.c cnijfilter-source/backendnet/lmonitor/cnijnetchk.c
--- cnijfilter-source-3.80-1/backendnet/lmonitor/cnijnetchk.c	2012-07-03 02:20:08.000000000 +0200
+++ cnijfilter-source/backendnet/lmonitor/cnijnetchk.c	2017-10-01 18:15:58.779928638 +0200
@@ -328,14 +328,12 @@
 static int getStatusCode(const char *src, const char *key, const char *val, const int index){
 	int i, j, start=-1;
 	int len = 0, keylen=0;
-	unsigned long bufsize=0; 
 	char result[1024];
 	char tmp[1024], *buf=NULL;
 	
 	if (src==NULL) return CNNL_RET_FAILURE;
 	if (key==NULL) return CNNL_RET_FAILURE;
 	
-	bufsize = (src[0] << 8) + src[1]; 
 	buf = (char*)&(src[2]);
 
 	len = strlen(buf);
diff -ru cnijfilter-source-3.80-1/cngpij/cngpij/bjcups.c cnijfilter-source/cngpij/cngpij/bjcups.c
--- cnijfilter-source-3.80-1/cngpij/cngpij/bjcups.c	2012-04-17 05:39:20.000000000 +0200
+++ cnijfilter-source/cngpij/cngpij/bjcups.c	2017-10-01 18:15:58.799928638 +0200
@@ -685,8 +685,8 @@
 					*pResponse;					// Pointer to CUPS IPP response.
 	ipp_attribute_t	*pAttribute;				// Pointer to CUPS attributes.
 	cups_lang_t		*pLanguage;					// Pointer to language.
-	char			*pPrinter = NULL;			// Pointer to printer name.
-	char			*pDUri = NULL;				// Pointer to Device uri.
+	const char		*pPrinter = NULL;			// Pointer to printer name.
+	const char		*pDUri = NULL;				// Pointer to Device uri.
 	short			retVal = -1;	// Return value.
 /*** Parameters end ***/
 	
@@ -698,8 +698,8 @@
 	else {
 		pRequest = ippNew();
 		
-		pRequest->request.op.operation_id = CUPS_GET_PRINTERS;
-		pRequest->request.op.request_id   = 1;
+		ippSetOperation(pRequest, CUPS_GET_PRINTERS);
+		ippSetRequestId(pRequest, 1);
 		
 		pLanguage = bjcupsLangDefault();	// cupsLangDefault() -> bjcupsLangDefault() for cups-1.1.19
 		
@@ -708,29 +708,29 @@
 		ippAddString(pRequest, IPP_TAG_OPERATION, IPP_TAG_URI, "printer-uri", NULL, NULL);
 		
 		if ((pResponse = cupsDoRequest(pHTTP, pRequest, "/")) != NULL) {
-			if (pResponse->request.status.status_code > IPP_OK_CONFLICT) {
+			if (ippGetStatusCode(pResponse) > IPP_OK_CONFLICT) {
 				fputs("ERROR: IPP ERROR\n", stderr);
 				goto onErr;
 			}
 			else {
-				pAttribute = pResponse->attrs;
+				pAttribute = ippFirstAttribute(pResponse);
 
 				while (pAttribute != NULL) {
-					while (pAttribute != NULL && pAttribute->group_tag != IPP_TAG_PRINTER) {
-						pAttribute = pAttribute->next;
+					while (pAttribute != NULL && ippGetGroupTag(pAttribute) != IPP_TAG_PRINTER) {
+						pAttribute = ippNextAttribute(pResponse);
 					}
 					if (pAttribute == NULL) {
 						break;
 					}
 					
-					while (pAttribute != NULL && pAttribute->group_tag == IPP_TAG_PRINTER) {
-						if (strcmp(pAttribute->name, "printer-name") == 0 && pAttribute->value_tag == IPP_TAG_NAME) {
-							pPrinter = pAttribute->values[0].string.text;
+					while (pAttribute != NULL && ippGetGroupTag(pAttribute) == IPP_TAG_PRINTER) {
+						if (strcmp(ippGetName(pAttribute), "printer-name") == 0 && ippGetValueTag(pAttribute) == IPP_TAG_NAME) {
+							pPrinter = ippGetString(pAttribute, 0, NULL);
 						}
-						if (strcmp(pAttribute->name, "device-uri") == 0 && pAttribute->value_tag == IPP_TAG_URI) {
-							pDUri = pAttribute->values[0].string.text;
+						if (strcmp(ippGetName(pAttribute), "device-uri") == 0 && ippGetValueTag(pAttribute) == IPP_TAG_URI) {
+							pDUri = ippGetString(pAttribute, 0, NULL);
 						}
-						pAttribute = pAttribute->next;
+						pAttribute = ippNextAttribute(pResponse);
 					}
 
 					if (strcasecmp(pDestName, pPrinter) == 0) {
@@ -739,7 +739,7 @@
 					}
 					
 					if (pAttribute != NULL)
-						 pAttribute = pAttribute->next;
+						pAttribute = ippNextAttribute(pResponse);
 				}
 			}
 			
diff -ru cnijfilter-source-3.80-1/cngpij/cngpij/getipc.c cnijfilter-source/cngpij/cngpij/getipc.c
--- cnijfilter-source-3.80-1/cngpij/cngpij/getipc.c	2012-03-29 06:50:26.000000000 +0200
+++ cnijfilter-source/cngpij/cngpij/getipc.c	2017-10-04 19:08:26.679990345 +0200
@@ -40,7 +40,7 @@
 	int server_fd;
 	int client_fd;
 	char buf[128];
-	size_t len;
+	socklen_t len;
 	short ret = RET_ERROR;
 
 	if( (server_fd = socket(PF_UNIX, SOCK_STREAM, 0)) < 0 )
diff -ru cnijfilter-source-3.80-1/cngpijmnt/src/getipc.c cnijfilter-source/cngpijmnt/src/getipc.c
--- cnijfilter-source-3.80-1/cngpijmnt/src/getipc.c	2012-03-29 06:50:26.000000000 +0200
+++ cnijfilter-source/cngpijmnt/src/getipc.c	2017-10-04 19:19:27.689988248 +0200
@@ -42,7 +42,7 @@
 	int server_fd;
 	int client_fd;
 	char buf[128];
-	size_t len;
+	socklen_t len;
 	short ret = RET_ERROR;
 
 	if( (server_fd = socket(PF_UNIX, SOCK_STREAM, 0)) < 0 )
diff -ru cnijfilter-source-3.80-1/cngpijmnt/src/main.c cnijfilter-source/cngpijmnt/src/main.c
--- cnijfilter-source-3.80-1/cngpijmnt/src/main.c	2012-04-26 11:49:34.000000000 +0200
+++ cnijfilter-source/cngpijmnt/src/main.c	2017-10-04 19:26:22.749986932 +0200
@@ -308,8 +308,8 @@
 					*pResponse;					// Pointer to CUPS IPP response.
 	ipp_attribute_t	*pAttribute;				// Pointer to CUPS attributes.
 	cups_lang_t		*pLanguage;					// Pointer to language.
-	char			*pPrinter = NULL;			// Pointer to printer name.
-	char			*pDUri = NULL;				// Pointer to Device uri.
+	const char		*pPrinter = NULL;			// Pointer to printer name.
+	const char		*pDUri = NULL;				// Pointer to Device uri.
 	short			retVal = -1;	// Return value.
 /*** Parameters end ***/
 	
@@ -321,8 +321,8 @@
 	else {
 		pRequest = ippNew();
 		
-		pRequest->request.op.operation_id = CUPS_GET_PRINTERS;
-		pRequest->request.op.request_id   = 1;
+		ippSetOperation(pRequest, CUPS_GET_PRINTERS);
+		ippSetRequestId(pRequest, 1);
 		
 		pLanguage = bjcupsLangDefault();	// cupsLangDefault() -> bjcupsLangDefault() for cups-1.1.19
 		
@@ -331,29 +331,29 @@
 		ippAddString(pRequest, IPP_TAG_OPERATION, IPP_TAG_URI, "printer-uri", NULL, NULL);
 		
 		if ((pResponse = cupsDoRequest(pHTTP, pRequest, "/")) != NULL) {
-			if (pResponse->request.status.status_code > IPP_OK_CONFLICT) {
+			if (ippGetStatusCode(pResponse) > IPP_OK_CONFLICT) {
 				fputs("ERROR: IPP ERROR\n", stderr);
 				goto onErr;
 			}
 			else {
-				pAttribute = pResponse->attrs;
+				pAttribute = ippFirstAttribute(pResponse);
 
 				while (pAttribute != NULL) {
-					while (pAttribute != NULL && pAttribute->group_tag != IPP_TAG_PRINTER) {
-						pAttribute = pAttribute->next;
+					while (pAttribute != NULL && ippGetGroupTag(pAttribute) != IPP_TAG_PRINTER) {
+						pAttribute = ippNextAttribute(pResponse);
 					}
 					if (pAttribute == NULL) {
 						break;
 					}
 					
-					while (pAttribute != NULL && pAttribute->group_tag == IPP_TAG_PRINTER) {
-						if (strcmp(pAttribute->name, "printer-name") == 0 && pAttribute->value_tag == IPP_TAG_NAME) {
-							pPrinter = pAttribute->values[0].string.text;
+					while (pAttribute != NULL && ippGetGroupTag(pAttribute) == IPP_TAG_PRINTER) {
+						if (strcmp(ippGetName(pAttribute), "printer-name") == 0 && ippGetValueTag(pAttribute) == IPP_TAG_NAME) {
+							pPrinter = ippGetString(pAttribute, 0, NULL);
 						}
-						if (strcmp(pAttribute->name, "device-uri") == 0 && pAttribute->value_tag == IPP_TAG_URI) {
-							pDUri = pAttribute->values[0].string.text;
+						if (strcmp(ippGetName(pAttribute), "device-uri") == 0 && ippGetValueTag(pAttribute) == IPP_TAG_URI) {
+							pDUri = ippGetString(pAttribute, 0, NULL);
 						}
-						pAttribute = pAttribute->next;
+						pAttribute = ippNextAttribute(pResponse);
 					}
 
 					if (strcasecmp(pDestName, pPrinter) == 0) {
@@ -362,7 +362,7 @@
 					}
 					
 					if (pAttribute != NULL)
-						 pAttribute = pAttribute->next;
+						pAttribute = ippNextAttribute(pResponse);
 				}
 			}
 			
diff -ru cnijfilter-source-3.80-1/cngpijmon/src/bjcupsmon_cups.c cnijfilter-source/cngpijmon/src/bjcupsmon_cups.c
--- cnijfilter-source-3.80-1/cngpijmon/src/bjcupsmon_cups.c	2012-05-22 11:49:27.000000000 +0200
+++ cnijfilter-source/cngpijmon/src/bjcupsmon_cups.c	2017-10-04 19:17:29.239988624 +0200
@@ -215,8 +215,8 @@
 	else {
 		pRequest = ippNew();
 		
-		pRequest->request.op.operation_id = IPP_GET_PRINTER_ATTRIBUTES;
-		pRequest->request.op.request_id   = 1;
+		ippSetOperation(pRequest, IPP_GET_PRINTER_ATTRIBUTES);
+		ippSetRequestId(pRequest, 1);
 		
 		pLanguage = bjcupsLangDefault();			// cupsLangDefault() -> bjcupsLangDefault() for cups-1.1.19 
 				
@@ -225,12 +225,12 @@
 		ippAddString(pRequest, IPP_TAG_OPERATION, IPP_TAG_URI, "printer-uri", NULL, pURI);
 
 		if ((pResponse = cupsDoRequest(pHTTP, pRequest, "/")) != NULL) {
-			if (pResponse->request.status.status_code > IPP_OK_CONFLICT) {
+			if (ippGetStatusCode(pResponse) > IPP_OK_CONFLICT) {
 				retVal = ID_ERR_CUPS_API_FAILED;
 			}
 			else {
 				if ((pAttribute = ippFindAttribute(pResponse, "printer-state", IPP_TAG_ENUM)) != NULL) {
-					printerState = (ipp_state_t)pAttribute->values[0].integer;
+					printerState = (ipp_state_t) ippGetInteger(pAttribute, 0);
 				}
 			}
 			
@@ -288,8 +288,8 @@
 		else {
 			pRequest = ippNew();
 			
-			pRequest->request.op.operation_id = IPP_GET_PRINTER_ATTRIBUTES;
-			pRequest->request.op.request_id   = 1;
+			ippSetOperation(pRequest, IPP_GET_PRINTER_ATTRIBUTES);
+			ippSetRequestId(pRequest, 1);
 			
 			pLanguage = bjcupsLangDefault();		// cupsLangDefault() -> bjcupsLangDefault() for cups-1.1.19
 			
@@ -298,7 +298,7 @@
 			ippAddString(pRequest, IPP_TAG_OPERATION, IPP_TAG_URI, "printer-uri", NULL, printerURI);
 			
 			if ((pResponse = cupsDoRequest(pHTTP, pRequest, "/")) != NULL) {
-				if (pResponse->request.status.status_code > IPP_OK_CONFLICT) {
+				if (ippGetStatusCode(pResponse) > IPP_OK_CONFLICT) {
 					retVal = ID_ERR_CUPS_API_FAILED;
 				}
 				else {
@@ -310,7 +310,7 @@
 					//}
 					pAttribute = ippFindAttribute(pResponse, "printer-state-message", IPP_TAG_TEXT);
 					if (pAttribute != NULL) {
-						strncpy(pStatus, pAttribute->values[0].string.text, bufSize);
+						strncpy(pStatus, ippGetString(pAttribute, 0, NULL), bufSize);
 					}
 				}
 				ippDelete(pResponse);
@@ -369,8 +369,8 @@
 			else {
 				pRequest = ippNew();
 				
-				pRequest->request.op.operation_id = IPP_CANCEL_JOB;
-				pRequest->request.op.request_id   = 1;
+				ippSetOperation(pRequest, IPP_CANCEL_JOB);
+				ippSetRequestId(pRequest, 1);
 				
 				pLanguage = bjcupsLangDefault();		// cupsLangDefault() -> bjcupsLangDefault() for cups-1.1.19
 				
@@ -381,7 +381,7 @@
 				ippAddString(pRequest, IPP_TAG_OPERATION, IPP_TAG_NAME, "requesting-user-name", NULL, cupsUser());
 				
 				if ((pResponse = cupsDoRequest(pHTTP, pRequest, "/jobs/")) != NULL) {
-					if (pResponse->request.status.status_code > IPP_OK_CONFLICT) {
+					if (ippGetStatusCode(pResponse) > IPP_OK_CONFLICT) {
 						retVal = ID_ERR_CUPS_API_FAILED;
 					}
 					ippDelete(pResponse);
@@ -420,7 +420,7 @@
 	cups_lang_t		*pLanguage;								// Pointer to language.
 	ipp_jstate_t	jobState = 0;							// Job state.
 	gint			jobID = 0;								// Job ID.
-	gchar			*pJobUserName = NULL;					// User name of print job.
+	const gchar			*pJobUserName = NULL;					// User name of print job.
 	uid_t			userID;									// User ID.
 	struct passwd	*pPasswd;								// Pointer to password structure.
 	gint			retVal = ID_ERR_PRINT_JOB_NOT_EXIST;	// Return value.
@@ -444,8 +444,8 @@
 	else {
 		pRequest = ippNew();
 		
-		pRequest->request.op.operation_id = IPP_GET_JOBS;
-		pRequest->request.op.request_id   = 1;
+		ippSetOperation(pRequest, IPP_GET_JOBS);
+		ippSetRequestId(pRequest, 1);
 		
 		pLanguage = bjcupsLangDefault();	// cupsLangDefault() -> bjcupsLangDefault() for cups-1.1.19
 		
@@ -456,31 +456,31 @@
 		ippAddStrings(pRequest, IPP_TAG_OPERATION, IPP_TAG_KEYWORD, "requested-attributes",(int)(sizeof(jobattrs) / sizeof(jobattrs[0])), NULL, jobattrs);
 		
 		if ((pResponse = cupsDoRequest(pHTTP, pRequest, "/")) != NULL) {
-			if (pResponse->request.status.status_code > IPP_OK_CONFLICT) {
+			if (ippGetStatusCode(pResponse) > IPP_OK_CONFLICT) {
 				retVal = ID_ERR_CUPS_API_FAILED;
 			}
 			else {
-				pAttribute = pResponse->attrs;
+				pAttribute = ippFirstAttribute(pResponse);
 
 				while (pAttribute != NULL) {
-					while (pAttribute != NULL && pAttribute->group_tag != IPP_TAG_JOB) {
-						pAttribute = pAttribute->next;
+					while (pAttribute != NULL && ippGetGroupTag(pAttribute) != IPP_TAG_JOB) {
+						pAttribute = ippNextAttribute(pResponse);
 					}
 					if (pAttribute == NULL) {
 						break;
 					}
 					
-					while (pAttribute != NULL && pAttribute->group_tag == IPP_TAG_JOB) {
-						if (strcmp(pAttribute->name, "job-id") == 0 && pAttribute->value_tag == IPP_TAG_INTEGER) {
-							jobID = pAttribute->values[0].integer;
+					while (pAttribute != NULL && ippGetGroupTag(pAttribute) == IPP_TAG_JOB) {
+						if (strcmp(ippGetName(pAttribute), "job-id") == 0 && ippGetValueTag(pAttribute) == IPP_TAG_INTEGER) {
+							jobID = ippGetInteger(pAttribute, 0);
 						}
-						if (strcmp(pAttribute->name, "job-state") == 0 && pAttribute->value_tag == IPP_TAG_ENUM) {
-							jobState = (ipp_jstate_t)pAttribute->values[0].integer;
+						if (strcmp(ippGetName(pAttribute), "job-state") == 0 && ippGetValueTag(pAttribute) == IPP_TAG_ENUM) {
+							jobState = (ipp_jstate_t) ippGetInteger(pAttribute, 0);
 						}
-						if (strcmp(pAttribute->name, "job-originating-user-name") == 0 && pAttribute->value_tag == IPP_TAG_NAME) {
-							pJobUserName = pAttribute->values[0].string.text;
+						if (strcmp(ippGetName(pAttribute), "job-originating-user-name") == 0 && ippGetValueTag(pAttribute) == IPP_TAG_NAME) {
+							pJobUserName = ippGetString(pAttribute, 0, NULL);
 						}
-						pAttribute = pAttribute->next;
+						pAttribute = ippNextAttribute(pResponse);
 					}
 					if (jobState == IPP_JOB_PROCESSING) {
 						if (pJobUserName != NULL) {
@@ -497,7 +497,7 @@
 					}
 
 					if (pAttribute != NULL)
-						pAttribute = pAttribute->next;
+						pAttribute = ippNextAttribute(pResponse);
 				}
 			}
 			
@@ -537,8 +537,8 @@
 					*pResponse;					// Pointer to CUPS IPP response.
 	ipp_attribute_t	*pAttribute;				// Pointer to CUPS attributes.
 	cups_lang_t		*pLanguage;					// Pointer to language.
-	gchar			*pPrinter = NULL;			// Pointer to printer name.
-	gchar			*pUri = NULL;				// Pointer to printer uri.
+	const gchar		*pPrinter = NULL;			// Pointer to printer name.
+	const gchar		*pUri = NULL;				// Pointer to printer uri.
 	gchar			*pTemp = NULL;				// Temporary pointer.
 	gint			i;							// Counter.
 	gint			retVal = ID_ERR_NO_ERROR;	// Return value.
@@ -555,8 +555,8 @@
 	else {
 		pRequest = ippNew();
 		
-		pRequest->request.op.operation_id = CUPS_GET_PRINTERS;
-		pRequest->request.op.request_id   = 1;
+		ippSetOperation(pRequest, CUPS_GET_PRINTERS);
+		ippSetRequestId(pRequest, 1);
 		
 		pLanguage = bjcupsLangDefault();	// cupsLangDefault() -> bjcupsLangDefault() for cups-1.1.19
 		
@@ -565,28 +565,28 @@
 		ippAddStrings(pRequest, IPP_TAG_OPERATION, IPP_TAG_KEYWORD, "requested-attributes", sizeof(attributes) / sizeof(attributes[0]), NULL, attributes);
 		
 		if ((pResponse = cupsDoRequest(pHTTP, pRequest, "/")) != NULL) {
-			if (pResponse->request.status.status_code > IPP_OK_CONFLICT) {
+			if (ippGetStatusCode(pResponse) > IPP_OK_CONFLICT) {
 				retVal = ID_ERR_CUPS_API_FAILED;
 			}
 			else {
-				pAttribute = pResponse->attrs;
+				pAttribute = ippFirstAttribute(pResponse);
 
 				while (pAttribute != NULL) {
-					while (pAttribute != NULL && pAttribute->group_tag != IPP_TAG_PRINTER) {
-						pAttribute = pAttribute->next;
+					while (pAttribute != NULL && ippGetGroupTag(pAttribute) != IPP_TAG_PRINTER) {
+						pAttribute = ippNextAttribute(pResponse);
 					}
 					if (pAttribute == NULL) {
 						break;
 					}
 					
-					while (pAttribute != NULL && pAttribute->group_tag == IPP_TAG_PRINTER) {
-						if (strcmp(pAttribute->name, "printer-name") == 0 && pAttribute->value_tag == IPP_TAG_NAME) {
-							pPrinter = pAttribute->values[0].string.text;
+					while (pAttribute != NULL && ippGetGroupTag(pAttribute) == IPP_TAG_PRINTER) {
+						if (strcmp(ippGetName(pAttribute), "printer-name") == 0 && ippGetValueTag(pAttribute) == IPP_TAG_NAME) {
+							pPrinter = ippGetString(pAttribute, 0, NULL);
 						}
-						if (strcmp(pAttribute->name, "printer-uri-supported") == 0 && pAttribute->value_tag == IPP_TAG_URI) {
-							pUri = pAttribute->values[0].string.text;
+						if (strcmp(ippGetName(pAttribute), "printer-uri-supported") == 0 && ippGetValueTag(pAttribute) == IPP_TAG_URI) {
+							pUri = ippGetString(pAttribute, 0, NULL);
 						}
-						pAttribute = pAttribute->next;
+						pAttribute = ippNextAttribute(pResponse);
 					}
 					
 					// Tora 020418: Compare two printer names ignoring the character case.
@@ -603,7 +603,7 @@
 					}
 
 					if (pAttribute != NULL)
-						 pAttribute = pAttribute->next;
+						 pAttribute = ippNextAttribute(pResponse);
 				}
 			}
 			
@@ -646,8 +646,8 @@
 					*pResponse;					// Pointer to CUPS IPP response.
 	ipp_attribute_t	*pAttribute;				// Pointer to CUPS attributes.
 	cups_lang_t		*pLanguage;					// Pointer to language.
-	gchar			*pPrinter = NULL;			// Pointer to printer name.
-	gchar			*pDUri = NULL;				// Pointer to Device uri.
+	const gchar		*pPrinter = NULL;			// Pointer to printer name.
+	const gchar		*pDUri = NULL;				// Pointer to Device uri.
 	gint			retVal = ID_ERR_NO_ERROR;	// Return value.
 /*** Parameters end ***/
 	
@@ -658,8 +658,8 @@
 	else {
 		pRequest = ippNew();
 		
-		pRequest->request.op.operation_id = CUPS_GET_PRINTERS;
-		pRequest->request.op.request_id   = 1;
+		ippSetOperation(pRequest, CUPS_GET_PRINTERS);
+		ippSetRequestId(pRequest, 1);
 		
 		pLanguage = bjcupsLangDefault();	// cupsLangDefault() -> bjcupsLangDefault() for cups-1.1.19
 		
@@ -668,28 +668,28 @@
 		ippAddString(pRequest, IPP_TAG_OPERATION, IPP_TAG_URI, "printer-uri", NULL, NULL);
 		
 		if ((pResponse = cupsDoRequest(pHTTP, pRequest, "/")) != NULL) {
-			if (pResponse->request.status.status_code > IPP_OK_CONFLICT) {
+			if (ippGetStatusCode(pResponse) > IPP_OK_CONFLICT) {
 				retVal = ID_ERR_CUPS_API_FAILED;
 			}
 			else {
-				pAttribute = pResponse->attrs;
+				pAttribute = ippFirstAttribute(pResponse);
 
 				while (pAttribute != NULL) {
-					while (pAttribute != NULL && pAttribute->group_tag != IPP_TAG_PRINTER) {
-						pAttribute = pAttribute->next;
+					while (pAttribute != NULL && ippGetGroupTag(pAttribute) != IPP_TAG_PRINTER) {
+						pAttribute = ippNextAttribute(pResponse);
 					}
 					if (pAttribute == NULL) {
 						break;
 					}
 					
-					while (pAttribute != NULL && pAttribute->group_tag == IPP_TAG_PRINTER) {
-						if (strcmp(pAttribute->name, "printer-name") == 0 && pAttribute->value_tag == IPP_TAG_NAME) {
-							pPrinter = pAttribute->values[0].string.text;
+					while (pAttribute != NULL && ippGetGroupTag(pAttribute) == IPP_TAG_PRINTER) {
+						if (strcmp(ippGetName(pAttribute), "printer-name") == 0 && ippGetValueTag(pAttribute) == IPP_TAG_NAME) {
+							pPrinter = ippGetString(pAttribute, 0, NULL);
 						}
-						if (strcmp(pAttribute->name, "device-uri") == 0 && pAttribute->value_tag == IPP_TAG_URI) {
-							pDUri = pAttribute->values[0].string.text;
+						if (strcmp(ippGetName(pAttribute), "device-uri") == 0 && ippGetValueTag(pAttribute) == IPP_TAG_URI) {
+							pDUri = ippGetString(pAttribute, 0, NULL);
 						}
-						pAttribute = pAttribute->next;
+						pAttribute = ippNextAttribute(pResponse);
 					}
 
 					if (strcasecmp(pDestName, pPrinter) == 0) {
@@ -698,7 +698,7 @@
 					}
 					
 					if (pAttribute != NULL)
-						 pAttribute = pAttribute->next;
+						 pAttribute = ippNextAttribute(pResponse);
 				}
 			}
 			
diff -ru cnijfilter-source-3.80-1/cngpijmon/src/bjcupsmon_main.c cnijfilter-source/cngpijmon/src/bjcupsmon_main.c
--- cnijfilter-source-3.80-1/cngpijmon/src/bjcupsmon_main.c	2012-03-27 03:36:03.000000000 +0200
+++ cnijfilter-source/cngpijmon/src/bjcupsmon_main.c	2017-10-01 18:15:59.049928637 +0200
@@ -371,14 +371,10 @@
 	} /* parents process */
 	else {
 		//for bscc
-		output_dev		Odev;
 	    char			buf[4096];
 
 		memset( buf, 0x00, sizeof(buf) ) ;
 
-		Odev.status = 0;
-		Odev.dev = 0;//device_type;	
-
 		/* Ver.3.70 */
 		if( (backend_Is == CANON_USB_BACKEND_CNIJ_USB) | (backend_Is == CANON_BJNP_BACKEND_CNIJ_NET) ){
 		//if( backend_Is == CANON_BJNP_BACKEND_CNIJ_NET ){
@@ -615,7 +611,6 @@
 	static struct timeb	prevTime;
 	static gboolean		initTime = FALSE;
 	//Ver.3.10 for Network
-	output_dev		Odev;
     char			buf[4096];
 	short			select_flag = 0;
 /*** Parameters end ***/
@@ -624,8 +619,6 @@
 	memset( buf, 0x00, sizeof(buf) ) ;
 	//20090202 inistialize
 	memset( statusStr, 0x00, sizeof(statusStr) ) ;
-	Odev.status = 0;
-	Odev.dev = 0;//device_type;	
 
 	if (initTime == FALSE) {
 		initTime = TRUE;
@@ -676,7 +669,6 @@
 				{
 					fd_set rfds;
 					struct timeval tv;
-					int retx = 0;
 					//int retval_select = 0;
 					//int retval;
 
@@ -694,7 +686,7 @@
 					//retval = select(1, &rfds, NULL, NULL, &tv);
 					//printf("###retval = %d\n",retval);
 					//if (retval > 0)
-						retx = read( 0, buf, sizeof(buf));
+						read( 0, buf, sizeof(buf));
 				}
 				/* read( 0, buf, sizeof(buf)); */
 
diff -ru cnijfilter-source-3.80-1/cnijfilter/src/bjfilter.c cnijfilter-source/cnijfilter/src/bjfilter.c
--- cnijfilter-source-3.80-1/cnijfilter/src/bjfilter.c	2012-03-29 03:22:47.000000000 +0200
+++ cnijfilter-source/cnijfilter/src/bjfilter.c	2017-10-04 19:15:55.149988922 +0200
@@ -29,6 +29,7 @@
 #include <popt.h>
 #include <time.h>
 #include <string.h>
+#include <ctype.h>
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/wait.h>
diff -ru cnijfilter-source-3.80-1/cnijfilter/src/bjfimage.c cnijfilter-source/cnijfilter/src/bjfimage.c
--- cnijfilter-source-3.80-1/cnijfilter/src/bjfimage.c	2012-03-22 02:50:19.000000000 +0100
+++ cnijfilter-source/cnijfilter/src/bjfimage.c	2017-10-01 18:15:59.079928637 +0200
@@ -39,6 +39,8 @@
 
 #include <stdio.h>
 #include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
 #include <png.h>
 
 #include "cpktypes.h"
@@ -1520,8 +1522,8 @@
 	short			tmpformat;
 	short			retbyte = 0;
 	short			bpp = 3;
-	long			width = 0;
-	long			length = 0;
+	png_uint_32		width = 0;
+	png_uint_32		length = 0;
 	long			rstep = 0;
 	long			RasterLength = 0;
 	long			i;
@@ -1574,7 +1576,7 @@
 		goto onErr;
 	}
 
-	if (setjmp (png_p->jmpbuf))
+	if (setjmp (png_jmpbuf(png_p)))
 	{
 		png_destroy_read_struct(&png_p, &info_p, (png_infopp)NULL);
 		goto onErr;
@@ -1585,7 +1587,7 @@
 
 	png_read_info( png_p, info_p );
 
-	png_get_IHDR( png_p, info_p, (unsigned long *)&width, (unsigned long *)&length, &bit_depth,
+	png_get_IHDR( png_p, info_p, &width, &length, &bit_depth,
 	              &color_type, &interlace_type, NULL, NULL);
 
 	/* not support Interlace */
diff -ru cnijfilter-source-3.80-1/cnijfilter/src/bjfoption.c cnijfilter-source/cnijfilter/src/bjfoption.c
--- cnijfilter-source-3.80-1/cnijfilter/src/bjfoption.c	2012-06-05 08:29:03.000000000 +0200
+++ cnijfilter-source/cnijfilter/src/bjfoption.c	2017-10-04 19:15:00.369989096 +0200
@@ -28,6 +28,7 @@
 #include <stdlib.h>
 #include <popt.h>
 #include <string.h>
+#include <ctype.h>
 
 #include "cncl.h"
 #include "cncldef.h"
diff -ru cnijfilter-source-3.80-1/cnijfilter/src/getipc.c cnijfilter-source/cnijfilter/src/getipc.c
--- cnijfilter-source-3.80-1/cnijfilter/src/getipc.c	2012-03-22 02:50:19.000000000 +0100
+++ cnijfilter-source/cnijfilter/src/getipc.c	2017-10-04 19:13:27.549989391 +0200
@@ -43,7 +43,7 @@
 	struct sockaddr_un		sun;
 	int						s, c;
 	char					buf[128];
-	size_t					adrlen;
+	socklen_t					adrlen;
 	short					ret = RET_ERROR;
 
 	if ((s = socket(PF_UNIX, SOCK_STREAM, 0)) < 0)
Seulement dans cnijfilter-source: .git
Seulement dans cnijfilter-source: .gitignore
Seulement dans cnijfilter-source: patch.sh
Seulement dans cnijfilter-source: .project
diff -ru cnijfilter-source-3.80-1/pstocanonij/filter/Makefile.am cnijfilter-source/pstocanonij/filter/Makefile.am
--- cnijfilter-source-3.80-1/pstocanonij/filter/Makefile.am	2012-03-29 06:50:26.000000000 +0200
+++ cnijfilter-source/pstocanonij/filter/Makefile.am	2017-10-01 18:15:59.099928637 +0200
@@ -1,4 +1,4 @@
-filterdir=$(libdir)/cups/filter
+filterdir=$(libexecdir)/cups/filter
 
 filter_PROGRAMS= pstocanonij
 
diff -ru cnijfilter-source-3.80-1/pstocanonij/filter/pstocanonij.c cnijfilter-source/pstocanonij/filter/pstocanonij.c
--- cnijfilter-source-3.80-1/pstocanonij/filter/pstocanonij.c	2012-06-26 01:55:26.000000000 +0200
+++ cnijfilter-source/pstocanonij/filter/pstocanonij.c	2017-10-01 18:15:59.109928637 +0200
@@ -388,7 +388,7 @@
 			{
 				if( !IS_BLANK(*p_code)  )
 					break;
-				*p_code++;
+				p_code++;
 			}
 			while( *p_code != '\0' )
 			{
@@ -1321,7 +1321,6 @@
 
 int exec_filter(char *cmd_buf, int ofd, int fds[2])
 {
-	int status = 0;
 	int	child_pid = -1;
 	char *filter_param[4];
 	char shell_buf[256];
@@ -1361,7 +1360,6 @@
 				execv(shell_buf, filter_param);
 						
 				fprintf(stderr, "execl() error\n");
-				status = -1;
 			}
 		}
 		else if( child_pid != -1 )
